### PR TITLE
reduced default guzzle set up

### LINKED ISSUE #13: how to set up guzzle
it seems to set up guzzle in a way that it throws exceptions when receiving a 403 response. that exception is correctly converted to a php-http exception, but still not expected if i understand correctly how we wanted things to be set up. (i did not add any plugins, so the plugin to convert error responses to exceptions is not there.)

it seems that we have the Middleware::httpErrors in the guzzle stack. looking at https://github.com/php-http/guzzle6-adapter/blob/master/src/Guzzle6HttpAdapter.php#L32 and https://github.com/guzzle/guzzle/blob/master/src/Client.php#L64-L65 i think we would need to provide an empty or reduced handler stack. this in turn means that using php-http results in a very bare-bone - but the alternative is inconsistent behaviour.

found this while working on https://github.com/FriendsOfSymfony/FOSHttpCache/pull/257 - to see a stack trace, look for `Http\Client\Exception\HttpException: Client error: `GET http://localhost:6181/user_context.php` resulted in a `403 Forbidden` response:` in the travis output.

### PR DESCRIPTION
**Pull request by dbu**
set up guzzle client as very minimal, not throwing exceptions for HTTP status codes, not following redirects, not handling cookies

fix #13

### CODE REVIEW

File: src/Guzzle6HttpAdapter.php [hunk 1]
```diff
@@ -28,7 +30,12 @@ class Guzzle6HttpAdapter implements HttpClient, HttpAsyncClient
      */
     public function __construct(ClientInterface $client = null)
     {
-        $this->client = $client ?: new Client();
+        if (!$client) {
+            $handlerStack = new HandlerStack();
+            $handlerStack->push(Middleware::prepareBody(), 'prepare_body');
```
Comment (dbu) [inline src/Guzzle6HttpAdapter.php]: as per #13, we keep the prepareBody functionality. contrary to redirect, exception and cookie, it does not change behaviour and there is no reason to not want this. also, if we add php-http plugins that handle content-length, this middleware realizes the value is already set and leaves things alone.

**Comment by sagikazarmark** [comment]
:+1:

**Comment by joelwurtz** [comment]
:+1:

**Comment by kbond** [comment]
This is actually causing me a problem. `Guzzle6HttpAdapter::sendRequest()` is returning `null`. When I look deeper into it, I see the guzzle promise is an instance of `GuzzleHttp\Promise\RejectedPromise` with reason: "No handler has been specified".

When I revert this change, it works as expected.

**Comment by sagikazarmark** [comment]
Fix is on it's way

**Comment by sagikazarmark** [comment]
@kbond can you try this fix: #21?

**Pull request merged**